docs/references/config/config.toml.md

```diff
@@ -0,0 +1,1749 @@
+last blocks. So, validators should stop the state machine, wait for some blocks, and then restart the state machine to
+avoid panic.
+
+### consensus.skip_timeout_commit
+Make progress as soon as we have all the precommits.
+```toml
+skip_timeout_commit = false
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `false` |
+|                     | `true`  |
+
+This is similar to setting [`consensus.timeout_commit`](#consensustimeout_commit) to `"0s"`.
+
+### consensus.create_empty_blocks
+If there are no transactions in the mempool, empty blocks are proposed to indicate that the chain is still running.
+```toml
+create_empty_blocks = true
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `true`  |
+|                     | `false` |
+
+This is more relevant to networks with a low number of transactions.
+
+### consensus.create_empty_blocks_interval
+If there are no transactions in the mempool, empty blocks are proposed to indicate that the chain is still running at
+this interval.
+```toml
+create_empty_blocks_interval = "0s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+### consensus.peer_gossip_sleep_duration
+Consensus reactor internal sleep duration when waiting for the next piece of calculation.
+```toml
+peer_gossip_sleep_duration = "100ms"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0ms"`     |
+
+This is a generic sleep duration for the consensus reactor that allows other threads to run when the reactor has no work
+to do.
+
+### consensus.peer_gossip_intraloop_sleep_duration
+Consensus reactor upper bound for a random sleep duration.
+```toml
+peer_gossip_intraloop_sleep_duration = "0s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`     |
+
+Random sleeps are inserted in the consensus reactor when it is waiting for HasProposalBlockPart messages and HasVote
+messages, so it can reduce the amount of these messages sent. This parameter sets an upper bound for the random value
+that is used for these sleep commands.
+
+### consensus.peer_query_maj23_sleep_duration
+Consensus reactor `queryMaj23Routine` function sleep time.
+```toml
+peer_query_maj23_sleep_duration = "2s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+Sleep time for the `queryMaj23Routine` function.
+
+## Storage
+Storage parameters are important in production settings as it can make the difference between a 2GB data folder and a
+20GB one.
+
+Storage pruning sets a flag in the CometBFT database for data that has expired. The database only deletes this data from
+the file system, when data compaction is called.
```

Comment:
```suggestion
CometBFT supports storage pruning to delete data indicated as not needed by the application or the data companion. Other than the pruning interval and compaction options, the configuration parameters in this section refer to the data companion. The applications pruning configuration is communicated to CometBFT via ABCI. 

Note that for some databases (GolevelDB), the data often does not get physically removed from storage due to the DB backend not triggering compaction. In these cases it is necessary to enable forced compaction and set the compaction interval accordingly. 
```

I will review this whole section later today but for now this statement is not 100% correct. 

Depending on the pruning interval, some physical deletions will happen. It will not happen enough to reflect what is desired by the pruning configuration (the storage will still grow). However, this seems to not be true for all DB backends either. In Pebble for example it seemed to actually delete the data as well.